src/tools/autosave.js

```diff
@@ -0,0 +1,99 @@
+import { exportJsmdToString } from './jsmd-tools'
+import { setPreviousAutosave } from '../actions/actions'
+
+function getAutosaveKey(state) {
+  const documentId = (state.notebookInfo.connectionMode === 'STANDALONE') ?
```

Comment:
This is throwing an error on my local version of the branch when I load a notebook from the server, for some reason. I believe you can use`connectionModeIsStandalone(state)` here, at any rate.